### PR TITLE
feat(query): make TransformBlockCompact respect  FUSE_OPT_KEY_BLOCK_IN_MEM_SIZE_THRESHOLD quota

### LINKED ISSUE #6910: Streaming load 60GB csv will OOM
Follow https://databend.rs/doc/learn/analyze-ontime-with-databend-on-ec2-and-s3

Start (release build) databend-query with local fs.

```shell
:( curl -H "insert_sql:insert into ontime format CSV" -H "skip_header:0" -H "field_delimiter:\t" -F "upload=@t_ontime.csv" -XPUT http://root:@127.0.0.1:8000/v1/streaming_load



curl: (56) Recv failure: Connection reset by peer
curl -H "insert_sql:insert into ontime format CSV" -H "skip_header:0" -H  -F   0.79s user 25.15s system 16% cpu 2:40.03 total
```

Databend Query

```shell
:) ./before-databend-query
Databend Query

Version: v0.7.146-nightly-94af6a2-simd(rust-1.64.0-nightly-2022-07-30T03:25:03.396932975Z)
Log:
    File: enabled level=INFO dir=./.databend/logs
    Stderr: disabled level=INFO
Meta: embedded at ./.databend/meta_embedded
Storage: fs://root=_data

MySQL
    listened at 127.0.0.1:3307
    connect via: mysql -uroot -h127.0.0.1 -P3307
Clickhouse(native)
    listened at 127.0.0.1:9000
    connect via: clickhouse-client --host 127.0.0.1 --port 9000
Clickhouse(http)
    listened at 127.0.0.1:8124
    usage:  echo 'create table test(foo string)' | curl -u root: '127.0.0.1:8124' --data-binary  @-
echo '{"foo": "bar"}' | curl -u root: '127.0.0.1:8124/?query=INSERT%20INTO%20test%20FORMAT%20JSONEachRow' --data-binary @-
Databend HTTP
    listened at 127.0.0.1:8000
    usage:  curl -u root: --request POST '127.0.0.1:8000/v1/query/' --header 'Content-Type: application/json' --data-raw '{"sql": "SELECT avg(number) FROM numbers(100000000)"}'

[1]    1085010 killed     ./before-databend-query
./before-databend-query  1672.18s user 2811.37s system 1291% cpu 5:47.12 total
```

Dmesg

```shell
[100202.445966] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/user@1000.service/app.slice/app-org.wezfurlong.wezterm-6c0843b02ff44de4974b3bf2e82f095a.scope,task=before-databend,pid=1085010,uid=1000
[100202.446149] Out of memory: Killed process 1085010 (before-databend) total-vm:76541232kB, anon-rss:39267488kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:111932kB oom_score_adj:200
[100205.025219] oom_reaper: reaped process 1085010 (before-databend), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

Runtime info:

```shell
:) neofetch
                   -`                    xuanwo@xuanwo-work
                  .o+`                   ------------------
                 `ooo/                   OS: Arch Linux x86_64
                `+oooo:                  Host: 90RC00EGCP LEGION REN7000P-26AMR
               `+oooooo:                 Kernel: 5.18.14-zen1-1-zen
               -+oooooo+:                Uptime: 1 day, 4 hours, 2 mins
             `/:-:++oooo+:               Packages: 1099 (pacman), 2 (nix-user)
            `/++++/+++++++:              Shell: zsh 5.9
           `/++++++++++++++:             Resolution: 3840x2160
          `/+++ooooooooooooo/`           DE: Plasma 5.25.3
         ./ooosssso++osssssso+`          WM: KWin
        .oossssso-````/ossssss+`         Theme: Breeze Light [Plasma], Breeze [GTK2/3]
       -osssssso.      :ssssssso.        Icons: [Plasma], breeze [GTK2/3]
      :osssssss/        osssso+++.       Terminal: WezTerm
     /ossssssss/        +ssssooo/-       CPU: AMD Ryzen 9 5950X (32) @ 3.400GHz
   `/ossssso+/:-        -:/+osssso+-     GPU: AMD ATI Radeon RX 6800/6800 XT / 6900 XT
  `+sso+:-`                 `.-/+oso:    Memory: 24989MiB / 64163MiB
 `++:.                           `-/+/
 .`                                 `/

```

---

cc @youngsofun, can you take a look?

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. make TransformBlockCompact respect FUSE_OPT_KEY_BLOCK_IN_MEM_SIZE_THRESHOLD quota.
2. remove `append_data` method in table.

Summary about this PR

Related #6910

### CODE REVIEW

File: query/src/pipelines/processors/transforms/transform_block_compact.rs [hunk 1]
```diff
@@ -101,12 +107,16 @@ impl Compactor for BlockCompactor {
         let block = blocks[size - 1].clone();
 
         // perfect block
-        if block.num_rows() >= self.min_row_per_block && block.num_rows() <= self.max_row_per_block
+        if (block.num_rows() >= self.min_row_per_block
+            && block.num_rows() <= self.max_row_per_block)
+            || block.memory_size() >= self.max_bytes_per_block
```
Comment (dantengsky) [inline query/src/pipelines/processors/transforms/transform_block_compact.rs:112]: seem a typo, `block.memory_size() <= self.max_bytes_per_block` instead?

Comment (sundy-li) [reply query/src/pipelines/processors/transforms/transform_block_compact.rs:112]: No, it means this block is large enough, so we should flush this out.

Comment (dantengsky) [reply query/src/pipelines/processors/transforms/transform_block_compact.rs:112]: oh, I see.

**Review by zhang2014 — APPROVED** [review]

**Review by dantengsky — APPROVED** [review]
LGTM!!

**Pull request merged**